packages/context/src/binding.ts

```diff
@@ -257,6 +274,22 @@ export class Binding {
   }
 
   /**
+   * Override any existing configuration on the context for
+   * this given binding and provide the given values for injection instead
+   * Any bindings not defined in the given dependencies will be resolved from
+   * the binding's context.
+   * @param deps
+   */
+  withDependencies(deps: BindingDependencies): this {
+    // This context is currently unbound.
```

Comment:
Any reason we don't bind context to parent here instead of doing it at getValue?